lib/browser/desktop-capturer.js

```diff
@@ -40,14 +40,14 @@ ipcMain.on(electronSources, (event, captureWindow, captureScreen, thumbnailSize,
   // If the WebContents is destroyed before receiving result, just remove the
   // reference from requestsQueue to make the module not send the result to it.
   event.sender.once('destroyed', () => {
-    request.webContents = null
+    request.event = null
   })
 })
 
 desktopCapturer.emit = (event, name, sources, fetchWindowIcons) => {
   // Receiving sources result from main process, now send them back to renderer.
   const handledRequest = requestsQueue.shift()
-  const handledWebContents = handledRequest.webContents
+  const handledWebContents = handledRequest.event ? handledRequest.event.sender : null
```

Comment:
since this is only actually used to switch on whether `event` is non-null, perhaps rather:

```js
const {event} = handledRequest
// ...
if (event) {
  event._replyInternal(...)
}
```